src/views/clusteroverview/OverviewTab/vm-statuses-card/utils/utils.ts

```diff
@@ -0,0 +1,73 @@
+import { V1VirtualMachine } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { getVMStatus } from '@kubevirt-utils/resources/shared';
+import { VM_STATUS } from '@kubevirt-utils/resources/vm/utils/vmStatus';
+import { RedExclamationCircleIcon } from '@openshift-console/dynamic-plugin-sdk';
+import { InProgressIcon, PausedIcon, SyncAltIcon } from '@patternfly/react-icons';
+
+import { ERROR } from './constants';
+
+const PRIMARY_STATUSES = [VM_STATUS.Running, VM_STATUS.Paused, VM_STATUS.Migrating, ERROR];
+
+const ERROR_STATUSES = [
+  VM_STATUS.CrashLoopBackOff,
+  VM_STATUS.ErrorUnschedulable,
+  VM_STATUS.ErrImagePull,
+  VM_STATUS.ImagePullBackOff,
+  VM_STATUS.ErrorPvcNotFound,
+  VM_STATUS.ErrorDataVolumeNotFound,
+  VM_STATUS.DataVolumeError,
+  VM_STATUS.Unknown,
+  VM_STATUS.WaitingForVolumeBinding,
+];
+
+export const vmStatusIcon = {
+  Running: SyncAltIcon,
+  Paused: PausedIcon,
+  Stopped: InProgressIcon,
+  Migrating: InProgressIcon,
+  Starting: InProgressIcon,
+  Stopping: InProgressIcon,
+  Deleting: InProgressIcon,
+  Provisioning: InProgressIcon,
+  Terminating: InProgressIcon,
+  Error: RedExclamationCircleIcon,
+};
+
+const initializeStatusCountsObject = (): { [key in VM_STATUS]?: number } =>
+  Object.keys(VM_STATUS).reduce((acc, status) => {
+    acc[status] = 0;
+    return acc;
+  }, {});
+
+export type StatusCounts = {
+  primaryStatuses: { [key in VM_STATUS | 'Error']?: number };
+  additionalStatuses: { [key in VM_STATUS]?: number };
+};
+
+export const getVMStatuses = (vms: V1VirtualMachine[]): StatusCounts => {
+  const statusCounts = initializeStatusCountsObject();
+
+  vms.forEach((vm) => {
+    const status = getVMStatus(vm);
+    statusCounts[status] = statusCounts[status] + 1;
+  });
+
+  const errorCount = ERROR_STATUSES.reduce(
+    (acc, state) => {
+      acc.Error = acc.Error + statusCounts?.[state] || 0;
+      delete statusCounts[state];
+      return acc;
+    },
+    { Error: 0 },
+  );
+
+  statusCounts[ERROR] = errorCount.Error;
```

Comment:
Can the `errorCount` be a number instead of an object?

Reply:
Yes and it makes it much cleaner. =) Thanks for catching that! It's been fixed.